# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/libwebrtcglue/VideoConduit.cpp
# Commit: d0b311007c03
# Full Hash: d0b311007c033e83824f5f6996a70ab9e870f31f
# Author: Byron Campen [:bwc] <docfaraday@gmail.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 5
# Description:
#   Bug 1654112 - Get RTCP BYE and RTP timeout handling working again (from Bug 1595479) r=mjf,dminor
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D106145
# ==============================================================================

diff -r daa9eac709dc -r d0b311007c03 dom/media/webrtc/libwebrtcglue/VideoConduit.cpp
--- a/dom/media/webrtc/libwebrtcglue/VideoConduit.cpp	Fri Feb 19 16:20:38 2021 +0000
+++ b/dom/media/webrtc/libwebrtcglue/VideoConduit.cpp	Fri Feb 19 15:56:41 2021 -0600
@@ -532,6 +532,7 @@
   }
 
   mRecvStreamConfig.decoder_factory = mDecoderFactory.get();
+  mRecvStreamConfig.rtp.rtcp_event_observer = this;
 
   mRecvStream =
       mCall->Call()->CreateVideoReceiveStream(mRecvStreamConfig.Copy());
@@ -1831,6 +1832,34 @@
   }
 }
 
+void WebrtcVideoConduit::OnRtcpBye() {
+  RefPtr<WebrtcVideoConduit> self = this;
+  NS_DispatchToMainThread(media::NewRunnableFrom([self]() mutable {
+    MOZ_ASSERT(NS_IsMainThread());
+    if (self->mRtcpEventObserver) {
+      self->mRtcpEventObserver->OnRtcpBye();
+    }
+    return NS_OK;
+  }));
+}
+
+void WebrtcVideoConduit::OnRtcpTimeout() {
+  RefPtr<WebrtcVideoConduit> self = this;
+  NS_DispatchToMainThread(media::NewRunnableFrom([self]() mutable {
+    MOZ_ASSERT(NS_IsMainThread());
+    if (self->mRtcpEventObserver) {
+      self->mRtcpEventObserver->OnRtcpTimeout();
+    }
+    return NS_OK;
+  }));
+}
+
+void WebrtcVideoConduit::SetRtcpEventObserver(
+    mozilla::RtcpEventObserver* observer) {
+  MOZ_ASSERT(NS_IsMainThread());
+  mRtcpEventObserver = observer;
+}
+
 bool WebrtcVideoConduit::HasCodecPluginID(uint64_t aPluginID) {
   MOZ_ASSERT(NS_IsMainThread());
 